action/protocol/staking/candidate_center_test.go

```diff
@@ -429,6 +429,93 @@ func TestFixAlias(t *testing.T) {
 	}
 }
 
+func TestMultipleNonStakingCandidate(t *testing.T) {
+	r := require.New(t)
+	candcenter, err := NewCandidateCenter(nil)
+	r.NoError(err)
+
+	candStaked := &Candidate{
+		Owner:              identityset.Address(1),
+		Operator:           identityset.Address(11),
+		Reward:             identityset.Address(3),
+		Name:               "self-staked",
+		Votes:              unit.ConvertIotxToRau(1200000),
+		SelfStakeBucketIdx: 1,
+		SelfStake:          unit.ConvertIotxToRau(1200000),
+	}
+	candNonStaked1 := &Candidate{
+		Owner:              identityset.Address(2),
+		Operator:           identityset.Address(12),
+		Reward:             identityset.Address(3),
+		Name:               "non-self-staked1",
+		Votes:              big.NewInt(0),
+		SelfStakeBucketIdx: candidateNoSelfStakeBucketIndex,
+		SelfStake:          big.NewInt(0),
+	}
+	candNonStaked2 := &Candidate{
+		Owner:              identityset.Address(3),
+		Operator:           identityset.Address(13),
+		Reward:             identityset.Address(3),
+		Name:               "non-self-staked2",
+		Votes:              big.NewInt(0),
+		SelfStakeBucketIdx: candidateNoSelfStakeBucketIndex,
+		SelfStake:          big.NewInt(0),
+	}
+	cands := []*Candidate{candStaked, candNonStaked1, candNonStaked2}
+	checkCandidates := func(candcenter *CandidateCenter, cands []*Candidate) {
+		r.True(testEqual(candcenter, CandidateList(cands)))
+	}
+	// add candidates
+	r.NoError(candcenter.Upsert(candStaked))
+	r.NoError(candcenter.Upsert(candNonStaked1))
+	r.NoError(candcenter.Upsert(candNonStaked2))
+	checkCandidates(candcenter, cands)
+	// commit
+	r.NoError(candcenter.Commit())
+	checkCandidates(candcenter, cands)
```

Comment:
just use `r.True(testEqual(candcenter, CandidateList(cands)))`? adding this func does not re-use code or reduce code duplication